### PR TITLE
Update sql_header docs

### PR DESCRIPTION
**Pull request by clrcrl**
## Description & motivation
We clarified this ourselves recently. Good to clarify it for the community

## To do
My read of the source code is that this is a config that overrides rather than accumulates (pre-hooks accumulate). Do we want to call this out explicitly?

## Pre-release docs
Is this change related to an unreleased version of dbt?
- [ ] Yes: please
    - update the base branch to `next`
    - add Changelog components: `<Changelog>[New/Changed] in v0.x.0</Changelog>`
    - add links to the "New and changed documentation" section of the latest [Migration Guide](../website/docs/docs/guides/migration-guide)
- [x] No: please ensure the base branch is `current`
- [ ] Unsure: we'll let you know!

### CODE REVIEW

File: website/docs/reference/resource-configs/sql_header.md [hunk 1]
```diff
@@ -2,25 +2,130 @@
...
+
+models:
+  +sql_header: "alter session set timezone = 'Australia/Sydney'"
+```
+
+</File>
+
+### Create a BigQuery Temporary UDF
+
+This example calls the `set_sql_header` macro. This macro is a convenience wrapper which you may choose to use if you have a multi-line SQL statement to inject. You do not need to use the `sql_header` configuration key in this case.
```
Comment (clrcrl) [inline website/docs/reference/resource-configs/sql_header.md:128]: it's still not clear to me why we provide this method. Why not just do:
```
{% set my_bq_udf %}
-- this is sql
{% endset %}

{{ config(sql_header = my_bq_udf) }}

```

Comment (jtcohen6) [reply website/docs/reference/resource-configs/sql_header.md:128]: No real reason, other than that's how the original contributor wanted to use it (https://github.com/fishtown-analytics/dbt/pull/1967)

**Review by jtcohen6 — APPROVED** [review]
Looks good!

What do you think of adding "when to use `pre-hook` and when to use `sql_header`" as an FAQ?

**Comment by clrcrl** [comment]
For now, I've got a "Comparison to pre-hooks" header which I think solves most of it. We can revisit later if we see the question popping up again :)

**Pull request merged**